### ISSUE TITLE
Disable Health Checks from Docker daemon to prevent conflict with liveliness probes.

### ISSUE THREAD

**Issue by ottoyiu** [kubernetes/kubernetes#50703]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:
Currently, if a Docker image has a 'HEALTHCHECK' instruction defined in the Dockerfile, Docker daemon continues to do the healthcheck. Duplicated health checks are the result of having it conflict with Kubernetes' liveliness checks.

**What you expected to happen**:
`--no-healthcheck=true` is passed in, on creation of the Docker container.

Reference:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/dockershim/docker_container.go#L122

https://github.com/kubernetes/kubernetes/blob/master/vendor/github.com/docker/docker/api/types/container/config.go#L49

**How to reproduce it (as minimally and precisely as possible)**:
Create a Dockerfile with a 'HEALTHCHECK' instruction; when doing a 'docker ps', you will see a Healthy or Unhealthy status on the running container.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.6.7
- Cloud provider or hardware configuration**:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by yujuhong** [comment]
@ottoyiu how does these checks interfere with kubernetes's liveness checks? Kubelet doesn't check the health status reported by docker.

ref: #25829

**Comment by ottoyiu** [comment]
@yujuhong It doubles the amount of health checks, which can lead to liveliness health checks to fail on applications that aren't as performant. Since the health status reported by Docker is not being consumed by Kubernetes, it's pretty useless.

**Comment by yujuhong** [comment]
Before we can properly support the HEALTHCHECK provided by docker images (discussed in #25829), I think it's reasonable to disable this by default.

/cc @yguo0905 for docker 1.13 validation

**Comment by dchen1107** [comment]
We need to disable docker's healthcheck before we figure out how to support it. Mark this for 1.8 milestone.

Since the health check was introduced in docker 1.12, we might have to patch Kubernetes 1.6, 1.7 if requires.

**Comment by dchen1107** [comment]
@yguo0905 and I introspected the feature earlier, especially with K8S integration with both docker 1.12.x and 1.13.x. I am summarizing what we found so far:

1) Docker daemon performs the health check in container's cgroup, not in daemon's cgroup. This is good. But the check consume the container's allocated resource, which might slow down the container's performance. 

2) There is a process leakage bug in this feature. We built an simple do-nothing image with the following HEALTHCHECK instruction:
```
HEALTHCHECK --interval=1s CMD bash -c 'while true; do true; done' || exit 1
``` 
The instruction has the following default options:
```
--interval=DURATION (default 30s)
--timeout=DURATION (default 30s)
--retries=N (default 3)
```
In above example, Docker just simple start a new healthcheck process in user's container every 30s without kill the previous processes. Of course, my example is pretty extreme, but it carries the idea. The good news is that stopping such container will end all those processes. 

3) We also create a good docker image but with a failed healthcheck on purpose to see if docker daemon's restart logic might interfere with Kubernetes's restart policy. The container is running happily, except through docker ps, the container is marked with unheathy. No unnecessary restart.
